### PR TITLE
Android streams implementation

### PR DESCRIPTION
**Pull request by kylelwm**
Created model folder with all model data inside
Implemented logout
Setup swipeRefreshLayout to list view for Android
Call and get stream data from server

### CODE REVIEW

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java [hunk 1]
```diff
@@ -1,31 +1,65 @@
...
      * number.
      */
     public static StreamRefreshListFragment newInstance(int sectionNumber) {
+
         StreamRefreshListFragment fragment = new StreamRefreshListFragment();
         Bundle args = new Bundle();
         args.putInt(ARG_SECTION_NUMBER, sectionNumber);
         fragment.setArguments(args);
+        fragment.sectionNumber = sectionNumber;
+        fragment.sectionNumber = sectionNumber;
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java]: I think there's a copy paste error?

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java [hunk 2]
```diff
@@ -35,8 +69,147 @@ public StreamRefreshListFragment() {
...
+        swipeRefreshLayout = (SwipeRefreshLayout) rootView.findViewById(R.id.swipeRefreshLayout);
+        listView = (ListView) rootView.findViewById(R.id.streamListView);
+        worldScopeStreamAdapter = new WorldScopeStreamAdapter(getActivity(), R.layout.fragment_stream_list_item, streams);
+        listView.setAdapter(worldScopeStreamAdapter);
+
+        // Set the onRefreshListener into swipeRefreshLayout
+        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                // Adds one more dummy when refresh
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java]: It doesn't add dummy already.

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeCreatedStream.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * Stream object returned from WorldScope App Server when creating a new stream
+ */
+public class WorldScopeCreatedStream extends WorldScopeStream {
+
+    private Date endedAt;
+    private String streamLink;
+
+    public Date getEndedAt() {return endedAt;}
+    public String getStreamLink() {return streamLink;};
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeCreatedStream.java]: Typo, additional semi-colon

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/WorldScopeAPIService.java [hunk 4]
```diff
@@ -1,26 +1,45 @@
 package com.litmus.worldscope;
 
+import com.litmus.worldscope.model.WorldScopeUser;
+import com.litmus.worldscope.model.WorldScopeViewStream;
+import java.util.List;
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/WorldScopeAPIService.java]: the import statements can be separated by line feeds

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeStream.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ * Generic Stream object model returned by WorldScope API Service
+ */
+public class WorldScopeStream {
+    private String streamId;
+    private String appInstance;
+    private String title;
+    private String roomId;
+    private int totalStickers;
+    private int totalViewers;
+    private boolean live;
```
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeStream.java:18]: Just a heads-up, there will be another `int totalComments`

**Comment by sharminelow** [comment]
Done the review. You can look at the comments.

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java (hunk 1 above)
Comment (kylelwm) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java]: Removed

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java (hunk 2 above)
Comment (kylelwm) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java]: Changed comment

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeCreatedStream.java (hunk 3 above)
Comment (kylelwm) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeCreatedStream.java]: Removed

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/WorldScopeAPIService.java (hunk 4 above)
Comment (kylelwm) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/WorldScopeAPIService.java]: Seperated

**Comment by sharminelow** [comment]
@kylelwm I think the changes haven't been pushed.

**Comment by kylelwm** [comment]
## New changes to original pull request

### Changelog

##### 1. Removed GIF player and use WebView to play GIF background instead

##### 2. Fix bug with pressing back button in main screen returning user to login screen

##### 3. Set up basic testing environment for Android client

##### 4. Wrote simple redirection tests for Android classes

##### 5. Updated JavaCV version and added filter for video to fix rotation issue

##### 6. Refactor and created several new fragments for streamer

###### \- Created a fragment for stream creation and termination

###### \- Created a fragment for stream controls

###### \- Created a fragment for actual streaming

##### 7. Added API end point for creating stream

##### 8. Redirect user to view screen when a stream is selected

##### 9. Refactor WorldScopeStream objects to be parcelable

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java (hunk 1 above)
Comment (sharminelow) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java]: The duplicate error is still present

Comment (kylelwm) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java]: Sorry, removed!

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java (hunk 2 above)
Comment (kylelwm) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/StreamRefreshListFragment.java]: Rechanged comment

File: client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeStream.java (hunk 5 above)
Comment (kylelwm) [inline client/android/WorldScope/app/src/main/java/com/litmus/worldscope/model/WorldScopeStream.java:18]: Noted

**Comment by kylelwm** [comment]
Please review =)

**Comment by sharminelow** [comment]
Ok, good to proceed.

**Comment by kylelwm** [comment]
Merged

**Pull request merged**